components/script/dom/htmlareaelement.rs

```diff
@@ -12,10 +12,132 @@ use dom::domtokenlist::DOMTokenList;
 use dom::htmlelement::HTMLElement;
 use dom::node::Node;
 use dom::virtualmethods::VirtualMethods;
+use euclid::point::Point2D;
 use std::default::Default;
 use string_cache::Atom;
 use style::attr::AttrValue;
 
+pub enum Area {
+    Circle { left: f32, top: f32, radius: f32 },
+    Rectangle { left_l: f32, top_t: f32, left_r: f32, top_b: f32 },
+    Polygon { points: Vec<f32> },
+    Default,
+}
+
+impl Area {
+    pub fn get_area (coord: String) -> Area
```

Comment:
Also, a comment with a spec link pointing to how this should be parsed would be helpful, I expect this function can become much simpler.